src/MessageTrait.php

```diff
@@ -204,6 +204,9 @@ public function withHeader($header, $value)
         $normalized = strtolower($header);
 
         $new = clone $this;
+        if($new->hasHeader($header)) {
+            unset($new->headers[$new->headerNames[$normalized]]);
```

Comment:
The problem there is, if  ```$new->headerNames[$normalized]```doesn't exist PHP will throw a warning.
I guess we could not use ```$new->hasHeader()``` and directly call ```array_key_exists($normilized, $new->headerNames)```.
What do you think?